system-addon/lib/SectionsManager.jsm

```diff
@@ -185,12 +189,46 @@ const SectionsManager = {
    * to false.
    *
    * @param options section options
+   * @param id      section ID
    */
-  updateLinkMenuOptions(options) {
+  updateLinkMenuOptions(options, id) {
     if (options.availableLinkMenuOptions) {
       options.contextMenuOptions = options.availableLinkMenuOptions.filter(
         o => !this.CONTEXT_MENU_PREFS[o] || Services.prefs.getBoolPref(this.CONTEXT_MENU_PREFS[o]));
     }
+
+    // Once we have rows, we can give each card it's own context menu based on it's type.
+    // We only want to do this for highlights because those have different data types.
+    // All other sections (built by the web extension API) will have the same context menu per section
+    if (options.rows && id === "highlights") {
+      this._addCardTypeLinkMenuOptions(options.rows);
+    }
+  },
+
+  /**
+   * Sets each card in highlights' context menu options based on the card's type.
+   * (See types.js for a list of types)
+   *
+   * @param rows section rows containing a type for each card
+   */
+  _addCardTypeLinkMenuOptions(rows) {
+    for (let card of rows) {
+      switch (card.type) {
+        case "history":
+          card.contextMenuOptions = CONTEXT_MENU_OPTIONS_FOR_HIGHLIGHT_TYPES.history;
```

Comment:
Can't this just be:
```js
for (let card of rows) {
  card.contextMenuOptions = CONTEXT_MENU_OPTIONS_FOR_HIGHLIGHT_TYPES[card.type];
}
```

You may also want to throw an error saying something like `No context menu for highlight type ${card.type} is configured` if you're not providing a default case.